app/controllers/facilities_controller.rb

```diff
@@ -75,12 +76,47 @@ def manage
     @active_tab = "admin_facility"
   end
 
+  require "zip"
+
+  #Start
+
+  # GET /facilities/:facility_id/thumbnail
+  def thumbnail
+    raise ActiveRecord::RecordNotFound if @facility.thumbnail.template.empty?
+    @file = @facility.thumbnail.new(file_type: "img")
+  end
+
+  # POST /facilities/:facility_id/upload_thumbnail
+  def upload_thumbnail
+    @file = @facility.thumbnail.new(params[:stored_file]&.permit(:file))
+    @file.file_type  = "img"
+    @file.name       = @facility+"_thumbnail"
+    @file.facility = @facility
```

Comment:
I think this should be already set because we're doing `thumbnail.new` off of `@facility`